deps/rabbitmq_mqtt/src/rabbit_mqtt_reader.erl

```diff
@@ -24,19 +24,21 @@
 
 -include("rabbit_mqtt.hrl").
 
+-type option(T) :: undefined | T.
+
 -define(HIBERNATE_AFTER, 1000).
 -define(PROTO_FAMILY, 'MQTT').
 
 -record(state,
-        {socket,
-         proxy_socket :: undefined | {rabbit_proxy_soket, any(), any()},
+        {socket :: rabbit_net:socket(),
+         proxy_socket :: option({rabbit_proxy_socket, any(), any()}),
          await_recv :: boolean(),
-         deferred_recv :: undefined | binary(),
-         parse_state,
+         deferred_recv :: option(binary()),
+         parse_state :: atom(),
          proc_state :: rabbit_mqtt_processor:state(),
          connection_state :: running | blocked,
          conserve :: boolean(),
-         stats_timer,
+         stats_timer :: option(reference()),
```

Comment:
The `stats_timer` is a record returned in https://github.com/rabbitmq/rabbitmq-server/blob/1fd4a6d353bd57793d38f7640bdfce1c6e6ada64/deps/rabbit_common/src/rabbit_event.erl#L95-L96 . Ideally, we should also declare the stats_timer to be opaque.

Reply:
done